### ISSUE TITLE
Granite worker does not remove stones at left upper corner of any building (bug from original game)

### ISSUE THREAD

**Issue by Wakeboardtiger** [nicymike/serflings#24]
https://github.com/nicymike/serflings/issues/14#issuecomment-666871843

**Comment by nicymike** [comment]
As I said, I will block building at a location down left of stone. It will be very hard to let the stonecutter enter the stone from a different location because there are no animations for this (as far as I know).

I also think the map generation makes sure the location is not blocked because the stonecutter only checks for buildings at that location when searching for stone to cut. If that location will be blocked by another stone for example, the stonecutter will go out and never reach that location.

I know this will remove a possible construction site but you can always build a stonecutters to remove the stone. 🙂
But I will also add a setting to disable the blocking.

**Comment by Wakeboardtiger** [comment]
> But I will also add a setting to disable the blocking.
That sounds like a fine solution. 
Thanks again for your great work.

**Comment by Spikeone** [comment]
Feels like roads and flags do also block, don't they?

**Comment by nicymike** [comment]
No, roads and flags do not block, because the stonecutter only needs to reach the spot. The map generator also makes sure that the spot is not blocked by anything not passable.

**Comment by Spikeone** [comment]
strange, one ignored a granite block, but when I changed the road he worked that one - maybe bad luck

**Comment by nicymike** [comment]
I guess so. Finding a spot only depends on luck.

**Comment by Richard-L** [comment]
I think it's fine if they just "walk over" whatever is on the bottom right.

BTW, demolishing "blocking" buildings does not make another stone mason tackle the then freed stone: 
![image](https://user-images.githubusercontent.com/4934209/107132006-cc523c00-68db-11eb-8127-82f4750ce751.png)

**Comment by nicymike** [comment]
Try waiting for a longer time. Searching for stone only depends on random search for locations. The stonecutter should find the stone eventually.

/cc @Richard-L

**Comment by samtigall** [comment]
This also heavily impacts the behaviour of the ai. In this level 30 scenario for example the enemy is not able to build any stone requiring buildings because he thinks he still has a stonecutter with stone nearby and he wont build any other stonecutter.
![grafik](https://github.com/user-attachments/assets/f5c35b27-87f2-42e2-b205-6d601d151caa)
[SAVE2.zip](https://github.com/user-attachments/files/16553601/SAVE2.zip)
